extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -55,12 +53,25 @@ class AmpBaseCarousel extends PreactBaseElement {
     );
 
     this.slide_ = parseInt(element.getAttribute('slide'), 10);
-    return dict({
-      'defaultSlide': this.slide_ || 0,
-      'onSlideChange': (index) => {
-        fireSlideChangeEvent(this.win, element, index, ActionTrust.HIGH);
-      },
-    });
+    props['defaultSlide'] = this.slide_ || 0;
```

Comment:
Shouldn't this be in `BaseElement['props']`?

Reply:
This shouldn't be observed by the Mutation Observer or cause re-render on value update. Can we use both `BaseElement['staticProps']` and `init`? Their comments seem to indicate that they are mutually exclusive:
https://github.com/ampproject/amphtml/blob/93b57606c5b4f03d848873ae2d5cc4ebeae31d02/src/preact/base-element.js#L794

Reply:
Can we move to `BaseElement.init()` then? I don't think this is AMP specific.

Reply:
I could see adding to `BaseElement['props']`:
`'defaultSlide': {attr: 'slide', type: 'number', default: '0'},`
but didn't previously because it seems needed for AMP side in `mutationObserverCallback`

@caroqliu are you saying the `mutationObserverCallback` isn't necessary?

Reply:
https://github.com/ampproject/amphtml/blob/93b57606c5b4f03d848873ae2d5cc4ebeae31d02/src/preact/base-element.js#L271-L277

It looks like they are not mutually exclusive, and `init` takes precedence over `staticProps`. If I move this to staticProps, is the `mutationObserverCallback` no longer necessary?

Reply:
I think the reason it's in `init` is to give it a default value? The `|| 0` part. But we should be able to move to `static` and set `default: 0` in the configuration.

Reply:
- `defaultSlide` is the initial value of `element.getAttribute('slide')` - it should not change after `init()` is called, and even if it does, it should not impact the Preact layer since it is only used in `useState(defaultSlide)`
- if the `slide` attribute on `amp-base-carousel` _does_ mutate after `init()`, that should cause a new render in the Preact layer by calling `api.goToSlide(attrValue)` - which it does in `mutationObserverCallback`.

All of the above is not AMP specific and should go to `BaseElement`. 

For additional context, these are initialized this way because we don't actually want `slide` to be a prop in `amp-base-carousel` - if the user wants to manually change the active slide index, they should use the imperative API. This is intended to mimic the controlled versus uncontrolled input model in React - if we actually respected the `slide` prop, we would not have slide transitions without the author implementing updating the `slide` value themselves by listening to `onSlideChange`.

Reply:
Moved all of this into `BaseElement`; PTAL